modules/nvidia_plugin/src/cuda_graph_topology_runner.cpp

```diff
@@ -3,56 +3,92 @@
 //
 
 #include "cuda_graph_topology_runner.hpp"
-#include "cuda/graph.hpp"
+
 #include "cuda/event.hpp"
-#include "cuda_profiler.hpp"
 
 namespace ov {
 namespace nvidia_gpu {
 
-CudaGraphTopologyRunner::CudaGraphTopologyRunner(const CreationContext& context, const std::shared_ptr<const ov::Model>& model)
-    : SubGraph(context, model) {
-    if (!IsCudaGraphCompatible())
-        throw CudaGraphIncompatible{"The topology is incompatible with CUDA graphs."};
+CudaGraphTopologyRunner::CudaGraphTopologyRunner(const CreationContext& context,
+                                                 const std::shared_ptr<const ov::Model>& model)
+    : orig_subgraph_{context, model} {
+    std::vector<SubGraph::ExecSequence> sequences;
+    SubGraph::ExecSequence currentSequence;
+    const auto& origSequence = orig_subgraph_.getExecSequence();
+    const auto totalSize = origSequence.size();
+    if (totalSize == 0) {
+        throw_ov_exception("ExecSequence size is 0");
+    }
+    bool isLastOpCompatible = origSequence[0]->IsCudaGraphCompatible();
+    currentSequence.push_back(origSequence[0]);
+    for (size_t i = 1; i < totalSize; ++i) {
+        const auto& op = origSequence[i];
+        if (op->IsCudaGraphCompatible() != isLastOpCompatible) {
+            isLastOpCompatible = !isLastOpCompatible;
+            sequences.emplace_back(std::move(currentSequence));
+            currentSequence.clear();
+        }
+        currentSequence.push_back(op);
+    }
+    sequences.emplace_back(std::move(currentSequence));
+
+    const auto& memoryManager = orig_subgraph_.memoryManager();
+    for (auto&& sequence : sequences) {
+        subgraphs_.emplace_back(context, model, std::move(sequence), memoryManager);
+    }
 }
 
 void CudaGraphTopologyRunner::Run(const InferenceRequestContext& context, const DeviceMemBlock& memoryBlock) const {
-    context.getCudaGraphContext().graphExec.value().launch(context.getThreadContext().stream());
+    const auto& stream = context.getThreadContext().stream();
+    std::size_t graphIndex = 0;
+    for (auto& subgraph : subgraphs_) {
```

Comment:
Done.
Hope that we understood you correctly about this property.
I added `number_of_cuda_graphs` property which gives the number of CUDA graphs used for the model.
It it not the same as the number of subgraphs, because the latter also includes subgraphs which don't use CUDA graphs.

Reply:
> Done. Hope that we understood you correctly about this property. I added `number_of_cuda_graphs` property which gives the number of CUDA graphs used for the model. It it not the same as the number of subgraphs, because the latter also includes subgraphs which don't use CUDA graphs.

It will be fine to add description of new property into the docs to avoid user misunderstanding.

Reply:
Done